core/src/main/java/com/graphhopper/util/PathSimplification.java

```diff
@@ -67,41 +67,42 @@ public PointList simplify() {
             boolean simplificationPossible = true;
             int nonConflictingStart = 0;
             int nonConflictingEnd = Integer.MAX_VALUE;
-            int toSimplifyIndex = -1;
-            int toShiftIndex = -1;
+            int listIndexToSimplify = -1;
+            int listIndexToShift = -1;
 
-            endIntervals = calculateEndIntervals(endIntervals, startIntervals, offset, listsToSimplify);
+            endIntervals = calculateEndIntervals(endIntervals, startIntervals, offsets, listsToSimplify);
 
             // Find the intervals to run a simplification, if possible, and where to shift
             for (int i = 0; i < listsToSimplify.size(); i++) {
                 if (startIntervals[i] >= nonConflictingEnd || endIntervals[i] <= nonConflictingStart) {
                     simplificationPossible = false;
                 }
                 if (startIntervals[i] > nonConflictingStart) {
-                    toSimplifyIndex = -1;
+                    listIndexToSimplify = -1;
                     nonConflictingStart = startIntervals[i];
                 }
                 if (endIntervals[i] < nonConflictingEnd) {
-                    toSimplifyIndex = -1;
+                    listIndexToSimplify = -1;
                     nonConflictingEnd = endIntervals[i];
                     // Remember the lowest endInterval
-                    toShiftIndex = i;
+                    listIndexToShift = i;
                 }
                 if (startIntervals[i] >= nonConflictingStart && endIntervals[i] <= nonConflictingEnd) {
-                    toSimplifyIndex = i;
+                    listIndexToSimplify = i;
                 }
             }
 
-            if (toSimplifyIndex >= 0 && simplificationPossible) {
+            if (listIndexToSimplify >= 0 && simplificationPossible) {
                 // Only simplify if there is more than one point
                 if (nonConflictingEnd - nonConflictingStart > 1) {
                     int removed = douglasPeucker.simplify(pointList, nonConflictingStart, nonConflictingEnd);
                     if (removed > 0) {
                         for (int i = 0; i < listsToSimplify.size(); i++) {
-                            reduceLength(listsToSimplify.get(i), offset[i], startIntervals[i], endIntervals[i] - removed);
-                            if (listsToSimplify.get(i).get(0) instanceof PathDetail) {
-                                List<PathDetail> pathDetails = listsToSimplify.get(i);
-                                for (int j = offset[i] + 1; j < pathDetails.size(); j++) {
+                            List<PathDetail> pathDetails = listsToSimplify.get(i);
```

Comment:
This looks wrong, it can be either InstructionList or a list of PathDetail. It works, but I think we should not leave it like that. WDYT?